### PR TITLE
Enable auto-flushing of output to fix `testfeed` details mode

### PR DESCRIPTION
**Pull request by marcphilipp**
## Overview

Prior to this PR, `ConsoleLauncher` created `PrintWriters` that wouldn't flush automatically when `println` is called which stopped the `testfeed` details mode from being useful. Moreover, it didn't initialized them with the right `Charsets`. Now, we rely on Picocli's initialization again which does both correctly.

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)

### CODE REVIEW

File: junit-platform-console/src/main/java/org/junit/platform/console/ConsoleLauncher.java [hunk 1]
```diff
@@ -30,9 +30,7 @@
 public class ConsoleLauncher {
 
 	public static void main(String... args) {
-		PrintWriter out = new PrintWriter(System.out);
-		PrintWriter err = new PrintWriter(System.err);
```
Comment (sormuras) [inline junit-platform-console/src/main/java/org/junit/platform/console/ConsoleLauncher.java:34]: Why not "just" use `new PrintWriter(System.out|.err, true);` here?

Comment (marcphilipp) [reply junit-platform-console/src/main/java/org/junit/platform/console/ConsoleLauncher.java:34]: Because theoretically, we'd also have to determine the correct encoding (which Picocli already does using the `sun.stdout.encoding` and `sun.stderr.encoding` system properties).

**Review by sormuras — APPROVED** [review]
`record OutputStreamConfig(PrintWriter out, PrintWriter err) { ... }` 🤓

**Comment by marcphilipp** [comment]
It crossed my mind, too. 🙂 But then we couldn't encapsulate the fields.
